### ISSUE TITLE
limit parallel to a max number of branches at the same time

### ISSUE THREAD

**Issue by kaifox** [molr/molr#86]
Some times, due to limitation on devices etc. It would be useful to limit the number of children running in parallel in a parallel forEach() branch. 

In the DSL this could look some

```
foreach().parallel(5);

foreach().parallel(Placeholder<Integer>);
```

**Comment by kaifox** [comment]
already done
